Orderbazaar - RET10 Logs

orderbazaar opened the pull request:
Hi @Jagannath-wits @sandeepshahi 

We have addressed all the issues related to the Buyer App side. Each step was double-checked during today’s [2nd April] 11 AM call with Dharmendar. Kindly review the comments for each step and merge the PR.

**Flow 1:** Succeeded

______________________________________________________________________________________

**Flow 2:** Succeeded

______________________________________________________________________________________

**Flow 3:**
          "report": {
            "on_select": {
                "priceErr": "Warning: Quoted Price in /on_select INR 215 does not match with the total price of items in /select INR 255 i.e price for the item mismatch in on_search and on_select"
            }
        },

**Comments:**
Upon comparison between /select and /on_select, the /select total price was INR 215, while /on_select showed INR 248.75 (including delivery charges). The utility seems to be displaying it incorrectly. Kindly validate and confirm. This was also verified with Dharmender during the 11 AM call.

______________________________________________________________________________________

Flow 4: Succeeded

______________________________________________________________________________________

**Flow 5:** Succeeded
        "report": {
            "on_status_rto_delivered/disposed": {
                "message/order.fulfillments/0/end/location": "Mismatch occurred while comparing 'RTO' fulfillment object with on_status_rto_delivered/disposed on key 'end/location'",
                "message/order.fulfillments/1/start/location": "Mismatch occurred while comparing 'Delivery' fulfillment object with on_status_rto_delivered/disposed on key 'start/location'"
            }
        },

**Comments:**
The required start and end location data is already present in the payload. Despite this, the error still occurs. This has been validated and confirmed with Dharmender via call and email.

______________________________________________________________________________________

**Flow 6:**

1. on_search_full_catalog_refresh: 
             CommentsNeeds to be fixed on the seller side.

2. "schemaErr0": "/message/order/quote/breakup/3/price/value Price value should be a number in string with upto 2 decimal places"
            **Comments:-** Prices are already in the correct format (two decimal places). We tried multiple approaches during the 11 AM call with Dharmender, including removing trailing decimals and rounding values. However, the issue persists. As advised by Dharmender, we will keep it as is and proceed for the submission.

3. "returnFulfillment": "Return fulfillment/tags/list is missing in update_reverse_qc"
        **Comments:-** The tags and list fields are present in the return fulfillment object. Dharmender has reviewed and confirmed this.

4. "invldQuoteTrailPrices": "quote_trail price and item quote price sum for on_update should be equal to the price as in on_confirm"
           **Comments:-** Two items were selected, and one was returned. Hence, the quote_trail reflects the price of the remaining item, which is expected behavior.

5. Missing settlement_details Error:
"message/order.payment/@ondc/org/settlement_details/0": "Missing payment/@ondc/org/settlement_details as compared to previous calls or not captured correctly i.e. {\"settlement_counterparty\":\"buyer\",\"settlement_phase\":\"refund\",\"settlement_type\":\"upi\",\"settlement_amount\":\"550\",\"settlement_timestamp\":\"2025-03-10T18:27:09.285Z\"}"

**Comments:** As per Dharmender’s guidance, this error can be ignored.

Jagannath-wits commented:
- select.payment object is incorrect please refer to the contract 
https://docs.google.com/document/d/1brvcltG_DagZ3kGr1ZZQk4hG4tze3zvcxmGV4NMTzr8/edit?pli=1&tab=t.0

"payment":{
             "type":"ON-ORDER",
             "collected_by":"BAP",
             "status":"PAID",
             "params":{
                "transaction_id":"707a027b-e60b-44b9-b4df-dc06441921dc",
                "amount":"215.00",
                "currency":"INR"
             }
          }
Only payment.type is required i.e also optional

- "on_select": {
"priceErr": "Warning: Quoted Price in /on_select INR 215 does not match with the total price of items in /select INR 255 i.e price for the item mismatch in on_search and on_select"
}

this is a valid error please check the price of the item in on_search and whats quoted in on_select i.e for item id 641 the price in on_search is 85rs and in on_select it is 45rs 

Similarly please recheck all the other issues or you can reach out to me via mail regarding false errors if any. @orderbazaar

orderbazaar commented:
Hi @Jagannath-wits @sandeepshahi 

We have addressed the issues mentioned above, including the fix for Flow 5. Kindly review the PR and proceed with the merge.

[](https://github.com/ONDC-Official/v1.2.0-logs/pull/3453)
